files/utils.cc

```diff
@@ -192,10 +193,9 @@ string get_system_path(const string &path) {
 	if (pos == pos2 && pos3 != string::npos) {
 		int num_chars = GetShortPathName(new_path.c_str(), nullptr, 0);
 		if (num_chars > 0) {
-			char *short_path = new char [num_chars + 1];
-			GetShortPathName(new_path.c_str(), short_path, num_chars + 1);
-			new_path = short_path;
-			delete [] short_path;
+			std::vector<char> short_path(num_chars + 1);
```

Comment:
Why `vector` instead of `string`?

Reply:
Pre-C++11 habit I guess; std::string was not guaranteed to return the underlying array this way back in the day.
Now it is so yes, string should be as good.